src/components/index-page/Loading.tsx

```diff
@@ -0,0 +1,45 @@
+'use client';
+
+import { Token } from '@/app/api/getToken';
+import { useSearchParams } from 'next/navigation';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+export const Loading = (props: { social: string }) => {
+  const params = useSearchParams();
+  const router = useRouter();
+  const code = params.get('code');
+
+  const getUsersToken = async () => {
+    try {
+      const response = await Token.getToken(code, props.social);
+      const setAc = localStorage.setItem('AT', response.accessToken);
+      const setRc = localStorage.setItem('RT', response.refreshToken);
```

Comment:
이렇게 한번에 여러개의 데이터를 localStorage에 저장하는 용도의 유틸 함수를 만들어봐도 좋을 것 같아요 🤗
```ts
const setStorageValues = (list: [string, any][]) => {
  list.forEach(([key, value]) => {
    localStorage.setItem(key, JSON.stringify(value))
  });
}

setStorageValues([
  ['accessToken', response.accessToken], 
  ['refreshToken', response.refreshToken]
]);
```
추가적으로 네이밍을 더 명확하게 짓는게 좋을 것 같아요 key로 저장하는 `AT`, `RT`는 너무 단축어라고 생각이드네요! 네이밍이 생각보다 길어지더라도 명확하게 짓는 것이 어떨까요~?
저라면 `accessToken`, `refreshToken` 온전한 네이밍으로 저장 할 것 같습니다!

추가적으로 setItem은 반환값이 없는데(undefined) `setAc`, `setRc`의 용도가 있을까요? 
- https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem#return_value

Reply:
이러한 유틸 함수 관리하는게 당장에 내 프로젝트에 많이 쓰이지 않는 다면 `오버 엔지니어링`이라고 느낄 수 있어요 🥹 (당연합니다)

하지만, 단순하게 코드를 나열하는게 아닌 `특정 기능을 하는 함수로 분리`하고, `추상화`해서 사용하는게 나중에 여러분의 코드 퀄리티를 높이는데 많은 도움이 될 거에요 😁

코드를 짤 때 다양한 시야로 접근하면 좋을 것 같아 아이디어 제공 차원으로 코멘트를 남겨봐요

Reply:
저번에 같이 대화하고, util함수를 어떤식으로 분리하고 확장하는지 까다로웠는데 아는 내용으로 보니깐 이해가 잘 되네요, 저도 프로젝트 진행할 때 활용해 보겠습니다😊

Reply:
@zzzRYT 👍 좋습니다! 리뷰어에 함께 묶여 있어서 그냥 함께 참조를 드렸는데 도움이 됐다니 다행이네요

@dmsdnWkd1234 @chamjin 다른분들도 어떻게 생각하는지 궁금해요 ~~! 😄 
추가로 질문에대한 답변도 해주시면 좋을 것 같아요!
Pull Request로 논의하는게 실무에서 자주 발생하기 때문에 이런식으로 이야기하는 것에 익숙해지는 것도 좋아요 ㅎㅎㅎㅎ

Reply:
@ssi02014 
@ssi02014 
AT와 RT라는 제가 보기에도 부적절한 네이밍인것 같습니다... 수정해야 될 것 같습니다. 이것 말고도 지금 와서 제가 짠 코드들을 보면 네이밍이 부적절한 부분이 많이 보여서 리팩토링때 전체적으로 고쳐야할 부분입니다 감사합니다 😔 

`setAc`,`setRc`는 처음에 소셜 로그인마다 토큰을 저장하는 로직이 달라야 하는줄 알았습니다 그래서 각 소셜 로그인별 토큰을 저장하는 함수를 만들고 거기서 돌려쓰려다가 그럴 필요가 없다는걸 알았는데 수정을 안하고 그냥 써서... 이것도 굳이 이럴 필요가 없는것 같습니다 수정해야 할 부분입니다 😢

선배님의 코멘트를 보고 정말 많은 것들을 느꼈습니다 선배님이나 다른 사람들이 리뷰를 해주는 것을 보면 제가 너무 편협한 시각에서 코드를 짜고 있다는 것이 느껴져서 이런 리뷰가 저한테는 더 넓은 시야를 가지는데 너무 도움이 많이 됩니다 😸 코드의 재사용성이나 동작의 예측 가능성 같은 것들을 고려해서 코드를 짜는 것이 너무나 중요하지만 잘 안되고 있고 어려운거 같습니다 😭 😭 😭

```suggestion
      const setAc = localStorage.setItem('AT', response.accessToken);
      const setRc = localStorage.setItem('RT', response.refreshToken);
```
해당 로직은 선배님이 주신 의견을 토대로 수정하는게 좋아보여서 그렇게 진행하려고 합니다 예상치 못한 오류에도 대응할 수 있고 의도나 목적을 가진 함수를 만들어서 다른 사람이 보기에도 제가 나중에 보더라도 얘가 무슨 일을 하는건지 더 명확하게 알아볼수 있는게 더 좋은 코드인것 같습니다. 👍 👍 👍 

앞으로는 네이밍에도 조금 더 신경써야 할거 같습니다 너무 과한 단축어는 지양하고 무슨 기능을 하는지 알 수 있게 명확하게 짓도록 노력하겠습니다 🥇 

직장 때문에 많이 바쁘실텐데 시간 내서 리뷰 남겨주셔서 정말 감사드립니다 😂 😂 😂

Reply:
@dmsdnWkd1234 @chamjin @zzzRYT 의견을 긍정적으로 봐주셔서 감사드려요!! 😄

코드를 작성 할 때 특정 요구사항과 상황이 매번 달라 질 수 있어 정말 딱 이거다! 라는 정답은 없지만, 넓은 시야를 갖추고 있어야 중요한 선택의 갈림길에서 좋은 선택을 할 수 있는 가능성이 열린다고 생각해요

그러기 때문에 다양한 사람들의 의견을 듣는 것과 이런 코드리뷰 문화가 정말 성장에 많은 도움을 줄거에요! 👍
모던 애자일 팀의 코드 리뷰 문화가 정말 활성화가 잘됐으면 하는 바램입니다 🤗

저도 이런 문화가 잘 정착할 수 있도록 종종 찾아와서 의견드리고 싶은 내용이 있으면 코멘트로 남겨볼게요 ! 👍